src/erlcloud_inspector.erl

```diff
@@ -0,0 +1,1155 @@
+-include("erlcloud_aws.hrl").
+
+
+%%% Library initialization.
+-export([configure/2, configure/3, configure/4, new/2, new/3, new/4, inspector_result_fun/1]).
+
+
+%%% Inspector API
+-export([add_attributes_to_findings/2, add_attributes_to_findings/3,
+         attach_assessment_and_rules_package/2, attach_assessment_and_rules_package/3,
+         create_application/2, create_application/3,
+         create_assessment/3, create_assessment/4, create_assessment/5,
+         create_resource_group/1, create_resource_group/2,
+         delete_application/1, delete_application/2,
+         delete_assessment/1, delete_assessment/2,
+         describe_cross_account_access_role/0, describe_cross_account_access_role/1,
+         delete_run/1, delete_run/2,
+         describe_application/1, describe_application/2,
+         describe_assessment/1, describe_assessment/2,
+         describe_finding/1, describe_finding/2,
+         describe_resource_group/1, describe_resource_group/2,
+         describe_rules_package/1, describe_rules_package/2,
+         describe_run/1, describe_run/2,
+         detach_assessment_and_rules_package/2, detach_assessment_and_rules_package/3,
+         get_assessment_telemetry/1, get_assessment_telemetry/2,
+         list_applications/0, list_applications/1, list_applications/2,
+         list_assessment_agents/1, list_assessment_agents/2, list_assessment_agents/3,
+         list_assessments/0, list_assessments/1, list_assessments/2,
+         list_attached_assessments/1, list_attached_assessments/2, list_attached_assessments/3,
+         list_attached_rules_packages/1, list_attached_rules_packages/2, list_attached_rules_packages/3,
+         list_findings/0, list_findings/1, list_findings/2,
+         list_rules_packages/0, list_rules_packages/1, list_rules_packages/2,
+         list_runs/0, list_runs/1, list_runs/2,
+         list_tags_for_resource/1, list_tags_for_resource/2, list_tags_for_resource/3,
+         localize_text/1, localize_text/2, localize_text/3,
+         preview_agents_for_resource_group/1, preview_agents_for_resource_group/2, preview_agents_for_resource_group/3,
+         register_cross_account_access_role/1, register_cross_account_access_role/2, register_cross_account_access_role/3,
+         remove_attributes_from_findings/2, remove_attributes_from_findings/3,
+         run_assessment/2, run_assessment/3,
+         set_tags_for_resource/2, set_tags_for_resource/3,
+         start_data_collection/1, start_data_collection/2,
+         stop_data_collection/1, stop_data_collection/2,
+         update_application/3, update_application/4,
+         update_assessment/3, update_assessment/4]).
+
+
+%%%------------------------------------------------------------------------------
+%%% Shared types
+%%%------------------------------------------------------------------------------
+-type inspector_opts() :: list().
+
+-type return_val() :: {ok, proplists:proplist()} | {error, term()}.
+
+-export_type([return_val/0]).
```

Comment:
this exported name is too generic. please name it `inspector_return` or smth to avoid potential clashes with libarary users space